test/extended/openstack/machinesets.go

```diff
@@ -0,0 +1,135 @@
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+		})
+
+		g.It("should not have leftovers ports", func() {
+
+			skipUnlessMachineAPIOperator(dc, clientSet.CoreV1().Namespaces())
+			g.By("fetching worker machineSets")
+			var networkClient *gophercloud.ServiceClient
+			var rawBytes []byte
+			var t machineproviderv1.OpenstackProviderSpec
+			var rclient runtimeclient.Client
+
+			networkClient, err := client(serviceNetwork)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			machineSets, err := listWorkerMachineSets(dc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			if len(machineSets) == 0 {
+				e2eskipper.Skipf("Expects at least one worker machineset. Found none.")
+			}
+			xconfig, err := config.GetConfig()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			rclient, err = runtimeclient.New(xconfig, runtimeclient.Options{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			err = machinev1.AddToScheme(scheme.Scheme)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = configv1.AddToScheme(scheme.Scheme)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			newMachinesetParams := framework.BuildMachineSetParams(rclient, 1)
+			rawBytes, err = json.Marshal(newMachinesetParams.ProviderSpec.Value)
+
+			err = json.Unmarshal(rawBytes, &t)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			t.ServerGroupName = ""
+			t.ServerGroupID = "boo"
+			newMachinesetParams.Name = "bogus1-worker"
+			s, err := json.Marshal(t)
+			newMachinesetParams.ProviderSpec.Value.Raw = s
+
+			o.Expect(err).NotTo(o.HaveOccurred())
+			ms, err := framework.CreateMachineSet(rclient, newMachinesetParams)
+
+			clusterID := ms.Spec.Template.ObjectMeta.Labels["machine.openshift.io/cluster-api-cluster"]
+
+			_, err = framework.GetMachineSet(rclient, ms.Name)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			//Delete the machineset
+			e2e.Logf("Waiting for 10 seconds before deleting machineset %v", ms.Name)
+			time.Sleep(time.Second * 10)
+			framework.DeleteMachineSets(rclient, ms)
+			_, err = framework.GetMachineSet(rclient, ms.Name)
+			o.Expect(errors.IsNotFound(err)).To(o.BeTrue())
+
+			networkName := clusterID + "-openshift"
+
+			networkListOpts := networks.ListOpts{Name: networkName}
+			networkAllPages, err := networks.List(networkClient, networkListOpts).AllPages()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			allNetworks, err := networks.ExtractNetworks(networkAllPages)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			networkID := allNetworks[0].ID
+			e2e.Logf("Network ID: %v", networkID)
+			portListOpts := ports.ListOpts{
+				NetworkID: networkID,
+			}
+
+			// Check if an orphaned port related to the delete machineset exists
+			allPages, err := ports.List(networkClient, portListOpts).AllPages()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			allPorts, err := ports.ExtractPorts(allPages)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			portID := ""
+			for _, port := range allPorts {
+				if strings.Contains(port.Name, newMachinesetParams.Name) {
```

Comment:
is the name of the port exactly a fragment of the machineset name or it should be equal?
If it's equal, we could filter the ports query by that name as well

Reply:
The ports contains the string but it's not exact